src/Microsoft.ML.Data/Commands/TrainCommand.cs

```diff
@@ -64,6 +64,9 @@ public sealed class Arguments : DataCommand.ArgumentsBase
             [Argument(ArgumentType.AtMostOnce, IsInputFileName = true, HelpText = "The validation data file", ShortName = "valid")]
             public string ValidationFile;
 
+            [Argument(ArgumentType.AtMostOnce, IsInputFileName = true, HelpText = "The test data file", ShortName = "test")]
+            public string TestFile;
```

Comment:
>TestFile [](start = 26, length = 8)

I'm not sure I like the idea of having a test file be an argument to train command...

Reply:
we could have it hidden I guess

---
In reply to: [232073278](https://github.com/dotnet/machinelearning/pull/1571#discussion_r232073278) [](ancestors = 232073278)

Reply:
What's your concern? The validation set can be used to tune training parameters, so it can't be used to compute test scores sometime. If user want to have a fair mechanism to observer the change of test score in the training phase, this is the only solution she can have.

Reply:
I think we can solve this better by supporting multiple validation sets, rather than giving the training algorithms access to the test set.

Reply:
No. validation set is different from test sets. Training algorithm can access information in validation set but should never touch the test set.

Reply:
Yes, exactly. One validation set for pruning, one validation set for measuring the effectiveness of the pruning strategy. The training algorithm doesn't get access to the test set.

Reply:
How could the dev of an algorithm know the difference between validation[0] and validation[1]? I prefer to use the name, test set, to warn algorithms not to touch it.

Reply:
I think the solution as presented is good enough

---
In reply to: [232358388](https://github.com/dotnet/machinelearning/pull/1571#discussion_r232358388) [](ancestors = 232358388)